lib/ansible/modules/windows/win_hostname.ps1

```diff
@@ -0,0 +1,54 @@
+#!powershell
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+########
+
+function Get-ComputerName() {
+    Get-Content env:computername
+    return
+}
+
+function Set-ComputerName($newComputerName) {
+    $systemInfo = Get-WmiObject Win32_ComputerSystem
+    $cmdResult = $systemInfo.Rename($newComputerName)    
+    if ($cmdResult.ReturnValue -ne 0) {         
+        Fail-Json $result "The computer name could not be set. Check that it is a valid name."                
+    }
+    return $cmdResult.ReturnValue
+}
+
+########
+
+$params = Parse-Args $args -supports_check_mode $true;
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$diff_mode = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+
+$result = New-Object psobject @{
+    changed = $false
+};
+
+If (-not $params.name.GetType)
+{
+    Fail-Json $result "missing required arguments: name"
```

Comment:
We prefer to read out the parameters at the top of the module, with options to test for this.
```powershell
$name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
```